### ISSUE TITLE
UnicodeDecodeError and its possible effects

### ISSUE THREAD

**Issue by paul-bell** [extremeheat/JSPyBridge#53]
Greetings, and thank you for this terrific library.

I've lately hit one, perhaps two, problems with Pythonia.

From JavaScript I am calling a function in a dynamically loaded Python script. I pass an array of strings argument to Python. I get this result:

`Error: Attempt to access 'gen_enrichment_object' failed. Python didn't respond in time (100000ms), look above for any Python errors. If no errors, the API call hung`

Looking 'above' for any errors, I see this:

```
Traceback (most recent call last):
  File "/data/hermes/node_modules/pythonia/src/pythonia/interface.py", line 50, in <module>
    com_io()
  File "/data/hermes/node_modules/pythonia/src/pythonia/interface.py", line 39, in com_io
    data = apiin.readline()
  File "/usr/lib64/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 202: ordinal not in range(128)
```
But please note that this 'above' error happens exactly once, but I continue to see the first error ("...didn't respond time...") in subsequent calls to the Python function. That is, it seems as if the one "can't decode" error _might_ be affecting all subsequent calls despite the fact that the input to the function is different with each call.

Thank you for your guidance on this problem.

**Comment by paul-bell** [comment]
Please note that this problem occurs only under Linux (CentOS 8), but not under Windows.

**Comment by extremeheat** [comment]
Thanks for the report. It looks like an issue with unicode encoding. Will investigate, but does seem odd if behavior varies by OS. Do you have a snippet of code that can reproduce this?

**Comment by paul-bell** [comment]
Thanks for getting back to me; much appreciated.

I am using pythonia 0.2.16. While this seems to work on Windows but not on Linux (CentOS 8), it's probably worth mentioning that the app runs containerized (Docker) on Linux, but is a regular process on Windows.

There's not much to the code. From JavaScript I am calling a function in a .py file and passing it an array object:
```
let pyFli = await python('../fliscripts/' + scriptName);
let result = await pyFli.gen_enrichment_object(nlpText);  
```
The Python script looks like this:
```
import jsonpickle
import stanza
import re

nlp = stanza.Pipeline('en', model_dir='/home/node/stanza_resources', processors={'ner': 'ontonotes'})

script_name = 'ner_fli'

def gen_enrichment_object(text):                                # ipy.js calls this function
    result = []  
    text_cleaned = [re.sub(r"[^a-zA-Z0-9]", " ", i) for i in text]
    for i in text_cleaned:
        print(i)

        doc = nlp(i)
        for entity in doc.entities:
            if entity.type == 'DATE' or entity.type == 'TIME' or entity.type == 'GPE' or entity.type == 'FAC' or entity.type == 'LOC':
                    if i not in result:
                        result.append(i)
                    result.append(entity.text + " " + " ," + " "+ entity.type)

    print(result)

    eo = {
            "name": script_name,                                
            "value": result,                                     
    }

    pickled = jsonpickle.encode(eo)                             # make JSON string
    return pickled
```
If I have things right, then I've marked interface.py in line 39 below:

```
    while True:
        data = apiin.readline()     # LINE 39
        if not data:
            break
        if data[0] != "{":
            continue
        j = json.loads(data)
        bridge.onMessage(j["r"], j["action"], j["ffid"], j["key"], j["val"]) 
```
Thanks again for you help.

Cordially,

Paul

**Comment by paul-bell** [comment]
Hi,

Have you had a chance to dig into this?

Thank you.

-Paul

**Comment by extremeheat** [comment]
Merged and released as 1.0.0, you can update with `npm install pythonia@1`
